### DISCUSSION TITLE
Transparent background

### DISCUSSION THREAD

**Discussion by jvherck** [dicebear/dicebear#202] [Ideas]
The possibilities with this api are truly amazing! However it would be very handy and cool to have an option for a transparent background. That way we could easily change background too after downloading the image from the link, or just have a cool transparent avatar.

Much appreciation for your time on this project and I hope this would be implemented :)

**Comment by FlorianKoerner** [comment]
Hi @jvherck.

Actually, the background of almost all avatars is transparent by default. Do you have an example?

**Reply by jvherck** [reply]
![image](https://user-images.githubusercontent.com/75937214/157864460-0e72cd71-d74f-45dd-9f68-c8cf54791dc0.png)

The background here is white, which is the default I think. Is there a way to make the background transparent? I've made the background transparent by replacing every pixel with a certain color by an RGBA pixel that's transparent, however it also makes pixels in the avatar itself transparent if it's got the same color. So it would be a great asset to have the possibility to put `transparent` as `backgroundColor` in the API url.

**Reply by jvherck** [reply]
Wait, I always see a white background even if I open it up, and I see in the svg file code that it has a white <rect/> background, yet now on GitHub I see this is transparent... however on Discord it does show a white background. This is kind of confusing for me...

**Reply by FlorianKoerner** [reply]
The API outputs the avatars with transparent background - no matter if SVG or PNG.

<img width="306" alt="Bildschirmfoto 2022-03-12 um 08 40 04" src="https://user-images.githubusercontent.com/647303/158008866-1a3bf5f1-147c-4df3-98b8-da34f6cc02d2.png">

https://codepen.io/FlorianKoerner/pen/yLpLyyL

## GitHub

GitHub sets a white background via CSS - but your uploaded image actually has a transparent background as well.

<img width="320" alt="Bildschirmfoto 2022-03-12 um 08 41 53" src="https://user-images.githubusercontent.com/647303/158008920-c66cbb8d-f8c0-444e-ab42-405d946ee5c0.png">

When I change the color in the CSS,...

<img width="325" alt="Bildschirmfoto 2022-03-12 um 08 42 41" src="https://user-images.githubusercontent.com/647303/158008944-68508636-41ab-46a7-ba42-0b527082f5f2.png">

<img width="514" alt="Bildschirmfoto 2022-03-12 um 08 42 57" src="https://user-images.githubusercontent.com/647303/158008952-d30eb897-6fb9-4786-ab9f-811499f731ec.png">

## Discord 

When I share a link to the API in Discord, the background is also transparent.

<img width="462" alt="Bildschirmfoto 2022-03-12 um 09 01 30" src="https://user-images.githubusercontent.com/647303/158009601-7d95b0e2-9af2-4c74-840b-5c6a6ea32105.png">

Even if I upload your example.

<img width="346" alt="Bildschirmfoto 2022-03-12 um 09 02 28" src="https://user-images.githubusercontent.com/647303/158009613-4b797e88-d0af-4f67-9ee6-7b3926fb9794.png">

-----

So, unfortunately, I can't reproduce your problem.

**Reply by jvherck** [reply]
I didn't it was possible to just change backgrounds of an png like that, I thought it's just an image. I guess it's a cool special feature of png that I didn't know of yet and that would be the reason why sometimes it's transparent and sometimes not.
Thank you for your extensive and clear answer! 😄
